kernel/src/main/java/org/kframework/parser/inner/kernel/EarleyParser.java

```diff
@@ -583,11 +583,11 @@ public ParserMetadata(String input, Scanner scanner, Source source, int startLin
     }
 
     // the list of tokens in the sentence
-    Scanner.Token[] words;
+    public Scanner.Token[] words;
```

Comment:
this should not be public because arrays are mutable. Same with the two arrays below. You need a getter that returns a Collections.unmodifiableList